bio-qiime-pipeline/main.py

```diff
@@ -21,41 +23,76 @@ def main(experiment: Experiment[SequenceDataset]):
     else:
         logging.info(">> [Microbiome analysis] Single-end reads detected")
 
-    #  Optional: Primer trimming
-    if experiment.parameters["forwardAdapter"] or experiment.parameters["reverseAdapter"]:
-        logging.info(">> [Microbiome analysis] Trimming primers based on provided adapter sequences with cutadapt")
-        initialDataset = primerTrimming(experiment.dataset, experiment, pairedEnd)
-        initialDataset.download()
-    else:
-        initialDataset = experiment.dataset
+    if useCache:
+        demultiplexedDataset = getCache(getCacheNameOne(experiment))
+        if demultiplexedDataset is not None:
+            demultiplexedDataset.download()
+            uploadCacheAsArtifact(demultiplexedDataset, experiment)
```

Comment:
this code is repeated, can it be extracted into a function which takes in the cache name as a parameter, downloads the dataset, and uploads it as artifact?